ckanext/versioning/logic/action.py

```diff
@@ -113,6 +170,15 @@ def dataset_version_create(context, data_dict):
         raise toolkit.ValidationError(
             'Version names must be unique per dataset'
         )
+    #TODO: Names like 'Version 1.2' are not allowed as Github tags
+    backend = _get_github_backend()
+    current_revision = backend.revision_list(dataset.name)[0]
```

Comment:
I think it will be better to just call `backend.fetch(dataset.name)`. You will get the latest revision.